edalize/symbiyosys.py

```diff
@@ -0,0 +1,297 @@
+parameters and the chparam command comes from ``vlogparam`` parameters.
+
+Similarly, write a files section like this:
+
+.. code::
+
+    [files]
+    {{files}}
+
+which will expand to
+
+.. code::
+
+    [files]
+    ../path/to/some/file1.sv
+    ../path/to/file2.sv
+
+If you need more control over the read commands at the start of the script
+section, there are some variables that can help you.
+
+  - ``{{chparam}}`` expands to the chparam command at the end of gen_reads as
+    shown above.
+
+  - ``{{flags}}`` expands to the flags (defines and includes) used for each
+    read line.
+
+  - ``{{src_files}}`` expands to a list with the basename of each source file.
+
+You can reproduce the example above with something like
+
+.. code::
+
+    [script]
+    {% for name in src_files %}
+    read -sv {{flags}} {{name}}
+    {% endfor %}
+    {{chparam}}
+
+    '''
+
+    argtypes = ['vlogdefine', 'vlogparam']
+
+    tool_options = {
+        'lists': {
+            # A list of tasks to run from the .sby file. Passed on the sby
+            # command line.
+            'tasknames': 'String'
+        }
+    }
+
+    def __init__(self, edam=None, work_root=None, eda_api=None):
+        super(Symbiyosys, self).__init__(edam, work_root, eda_api)
+
+        # Register Jinja filters
+        self.jinja_env.filters["gen_reads"] = self._gen_reads
+
+        # The list of RTL paths in the fileset (populated at configure time by
+        # _get_file_names)
+        self.rtl_paths = None
+
+        # The list of include directories in the fileset (populated at
+        # configure time by _get_file_names)
+        self.incdirs = None
+
+        # The name of the interpolated .sby file that we create in the work
+        # root
+        self.sby_name = 'test.sby'
+
+    @staticmethod
+    def get_doc(api_ver):
+        if api_ver == 0:
+            return {'description':
+                    'SymbiYosys formal verification wrapper for Yosys',
+                    'lists': [
+                        {
+                            'name': 'tasknames',
+                            'type': 'String',
+                            'desc': ("A list of the .sby file's tasks to run. "
+                                     "Passed on the sby command line.")
+                        }
+                    ]}
+
+    def _get_file_names(self):
+        '''Read the fileset to get our file names'''
+        assert self.rtl_paths is None
+
+        src_files, self.incdirs = self._get_fileset_files()
+        self.rtl_paths = []
+        bn_to_path = {}
+        sby_names = []
+
+        # RTL files have types verilogSource or systemVerilogSource*. We
+        # presumably want some of them. The .sby file has type sbyConfig: we
+        # want exactly one of them.
+        for file_obj in src_files:
+            ft = file_obj.file_type
+            if ft.startswith('verilogSource') or ft == 'systemVerilogSource':
```

Comment:
```suggestion
            if ft.startswith('verilogSource') or ft.startsWith('systemVerilogSource'):
```

(These file types are a superset of the IP-XACT file types. IP-XACT specifies systemVerilogSource, systemVerilogSource-3.0, systemVerilogSource-3.1, and systemVerilogSource-3.1a. [Yes, super old.])

Reply:
Done! (With a regex, which seemed a bit neater than a "double startswith".)